core/tests/performance/smart_pool/src/smart_pool_benchmark.cpp

```diff
@@ -127,6 +127,36 @@ BENCHMARK_DEFINE_F(SmartPoolFixture, VectorMultiThreadOperate)(benchmark::State
     }
 }
 
+static void BM_SmartPoolCreateAndAcquireVector(benchmark::State &state)
+{
+    // Perform setup here
+    const int poolSize = 1;
+    const int vectorSize(state.range(0));
+
+    for (auto _ : state) {
+        kpsr::SmartObjectPool<std::vector<float>> smartPool("SmartPoolBenchmark",
```

Comment:
This should be outside the loop.

Reply:
But then the allocation is not taken into account for the time calculations. The idea was to compare it to std::vector<float>() + std::vector<float>.resize()